.github/workflows/testoperator_dispatch.yml

```
@@ -77,7 +87,7 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SPICED_COMMIT: ${{ steps.setup-spiced.outputs.SPICED_COMMIT }}
-          WORKFLOW_COMMIT: ${{ github.ref }}
+          WORKFLOW_COMMIT: ${{ matrix.commit }}
```

```suggestion
          WORKFLOW_COMMIT: ${{ github.ref }}
```

This needs to remain the ref of the executing dispatch workflow. Changing this to spiced commit will make dispatch execute on e.g. `trunk`, but the benchmark executes on a spiced commit HEAD.

Often, spiced commit will also differ from a fix branch where only changes to testoperator have been made. This also needs to remain `github.ref` for these cases.

---

I think for simplicity we should make the `matrix.commit` the same commit as the testoperator and the spiced under test. Trying to support a different testoperator version doesn't work in practice (i.e. snapshots being updated). Concretely this means we need to backport any relevant testoperator changes to the branch under test. Once we've gotten testoperator to be reliable, it shouldn't have too many changes.